### PR TITLE
parser: use the project loader code to find the origin's snapcraft.yaml

### PR DESCRIPTION
**Pull request by josepht**
Rather than roll our own snapcraft.yaml logic use the existing code in project_loader.

LP: #1665007

### CODE REVIEW

File: snapcraft/internal/parser.py [hunk 1]
```diff
@@ -51,9 +51,16 @@
...
-from snapcraft.internal import log, repo, sources
-from snapcraft.internal.errors import SnapcraftError, InvalidWikiEntryError
-from snapcraft.internal.project_loader import replace_attr
+from snapcraft.internal import log, project_loader, repo, sources
+from snapcraft.internal.errors import (
+    SnapcraftError,
+    InvalidWikiEntryError,
+    SnapcraftEnvironmentError,
+)
+from snapcraft.internal.project_loader import (
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: why do you import project_loader above and then do this finer grained import instead of using `project_loader`?

**Comment by josepht** [comment]
I'm not sure what you mean by 'formatted correctly'.

**Comment by kyrofa** [comment]
> I'm not sure what you mean by 'formatted correctly'.

- For the bugref, gbp-dch will be looking for LP:<space>#<bugnum>
- For the PR/commit title, we've starting following the convention of structuring it as: <effected subsystem>: <what changed>. So this might be "parser: use project loader to find origin's snapcraft.yaml" or something similar. This allows for one to extract meaning from `git log --oneline`, for instance.

**Comment by josepht** [comment]
Kyle: Thanks for explaining :)

FIxed.

**Comment by sergiusens** [comment]
Ok, better, a test would be nice though

File: integration_tests/snap.snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+description: A test part with snap/snapcraft.yaml
+maintainer: John Doe <john.doe@example.com>
+parts: [snap-testpart]
+origin: lp:~joetalbott/+junk/snap-testpart
```
Comment (kyrofa) [inline integration_tests/snap.snapcraft.yaml]: Does this make anyone else a tad nervous? I know this is the pattern followed elsewhere in the parser, where everything seems to be in @josepht's junk... but at least in my own experience, that's only for throw-away stuff. If this is accidentally deleted our tests start failing.

Is there a better place to put this stuff?

Comment (josepht) [inline integration_tests/snap.snapcraft.yaml]: I agree that it's suboptimal.  I'm working on an update to move all the parser tests to use git repos generated on-the-fly on the local disk.  Should speed up integration tests a bit too.

Comment (josepht) [inline integration_tests/snap.snapcraft.yaml]: I've pushed an update now.

**Comment by sergiusens** [comment]
nice on the locals, but how about you put all the wiki stubs in a `wiki` directory, kind of like what is done with `keys`

**Pull request merged**